barman/recovery_executor.py

```diff
@@ -1807,13 +1810,327 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+        )
+
+    def _prepare_destination(self, dest_dir):
+        """
+        Prepare the destination of a backup data/tablespace.
+
+        This method is responsible for removing a directory if it already
+        exists and for ensuring the correct permissions on the directory
+
+        :param str dest_dir: destination directory
+        """
+        # Remove a dir if exists. Ignore eventual errors
+        shutil.rmtree(dest_dir, ignore_errors=True)
+        # create the dir
+        mkpath(dest_dir)
+        # Ensure the right permissions to the destination directory
+        # chmod 0700 octal
+        os.chmod(dest_dir, 448)
+
+    def _retry_handler(self, dest_dirs, attempt):
+        """
+        Handler invoked during a combine backup in case of retry.
+
+        The method simply warn the user of the failure and
+        remove the already existing directories of the backup.
+
+        :param list[str] dest_dirs: destination directories
+        :param int attempt: attempt number (starting from 0)
+        """
+        output.warning(
+            "Failure combining backups using pg_combinebackup (attempt %s)", attempt
+        )
+        output.warning(
+            "The files created so far will be removed and "
+            "the combine process will restart in %s seconds",
+            "30",
+        )
+        # Remove all the destination directories and reinit the backup
+        for d in dest_dirs:
+            self._prepare_destination(d)
+
+    def _fetch_remote_status(self):
+        """
+        Gather info from the remote server.
+
+        This method does not raise any exception in case of errors,
+        but set the missing values to None in the resulting dictionary.
+        """
+        remote_status = dict.fromkeys(
+            (
+                "pg_combinebackup_installed",
+                "pg_combinebackup_tbls_mapping",
+                "pg_combinebackup_path",
+                "pg_combinebackup_version",
+            ),
+            None,
+        )
+
+        # Test pg_combinebackup existence
+        version_info = PgCombineBackup.get_version_info(self.server.path)
+
+        if version_info["full_path"]:
+            remote_status["pg_combinebackup_installed"] = True
+            remote_status["pg_combinebackup_path"] = version_info["full_path"]
+            remote_status["pg_combinebackup_version"] = version_info["full_version"]
+            pgcombinebackup_version = version_info["major_version"]
+        else:
+            remote_status["pg_combinebackup_installed"] = False
+            return remote_status
+
+        pg_version = None
+        if self.server.postgres is not None:
+            pg_version = self.server.postgres.server_major_version
+
+        # Is the tablespace mapping option supported?
+        if pgcombinebackup_version >= "9.4":
+            remote_status["pg_combinebackup_tbls_mapping"] = True
+        else:
+            remote_status["pg_combinebackup_tbls_mapping"] = False
```

Comment:
`pg_combinebackup` was introduced on Postgres 17, and there is no version without support for tablespace relocation.

I think we can get rid of `pg_combinebackup_tbls_mapping`.